flow-server/src/main/java/com/vaadin/flow/dom/Element.java

```diff
@@ -954,6 +994,14 @@ public double getProperty(String name, double defaultValue) {
                     return Double.NaN;
                 }
             }
+        } else if (value instanceof NumericNode) {
+            return ((NumericNode) value).asDouble(Double.NaN);
+        } else if (value instanceof BooleanNode) {
+            return ((BooleanNode) value).booleanValue() ? 1 : 0;
+        } else if (value instanceof TextNode) {
+            return ((TextNode) value).asDouble(Double.NaN);
+        } else if (value instanceof JsonNode) {
+            return ((JsonNode) value).asDouble(Double.NaN);
```

Comment:
nit: can use instance of pattern matching to improve readability.
Perhaps it can also be simplified by reducing to two cases: `BooleanNode` and `JsonNode` since or other Jackson type the handling is the same as for `JsonNode`

```suggestion
        } else if (value instanceof BooleanNode node) {
            return node.booleanValue() ? 1 : 0;
        } else if (value instanceof JsonNode node) {
            return node.asDouble(Double.NaN);
```

Reply:
done